src/CommandingOfficers/Venge.java

```diff
@@ -120,22 +121,22 @@ public void changeCombatContext(CombatContext instance)
   }
 
   @Override
-  public void applyCombatModifiers(BattleParams params, boolean amITheAttacker)
+  public void buffStrike(StrikeParams params)
   {
-    if( amITheAttacker )
-    {
       if( counterAtFullPower && params.isCounter )
       {
         // counterattack as if the unit had not taken damage.
-        params.attackerHP = params.attacker.getHP();
+        params.attackerHP = params.attacker.body.getHP();
```

Comment:
Any reason not to give `Combatant` a `getHP()` just to avoid this extra level of indirection?

Reply:
Not really.

Reply:
Actually, nevermind that.
Combatant is composed of `final` fields, and I think I want it to stay that way. If I was to put a `getHP()` on Combatant, I'd expect it to replace the instance in `params`, and I'm not gonna do that because then it'd have to be mutable.

Reply:
What? I'm confused. `getHP()` is not a mutator, and `params` doesn't have a `getHP` call.

Reply:
Yes, but `Combatant` exists as a handy summary-object of how a given unit will participate in a fight, for use in `Strike`/`BattleParams`.
You'll note I moved the `Unit`s, their `WeaponModel`s, and their positions into it and made them final to enforce not mucking about with that stuff.

By contrast, `Strike`/`BattleParams` has an HP value for each `Unit`, and it's important that it be mutable. If I put knowledge of HP state into the `Combatant`, I would expect that to represent the HP that would actually be used to calculate effective stats for the combat (and would also remove those members from the `params`).
By making the user go to the unit itself to fetch the HP, I'm making him acknowledge that he's looking at the *current board state* rather than what the combat will be calculated with.

Reply:
> I'm making him acknowledge that he's looking at the current board state

Hm. I'm not sure that distinction is as clear as you imagine it to be.

> You'll note I moved the Units, their WeaponModels, and their positions into it and made them final 

I hadn't. And if I'm just looking for an accessor (How much health does this guy have, anyway), I'm not going to look closely enough to see that the unit is declared `final` and then make generalizations about what that means about the state of the world.

Maybe add the utility method with a name like `get[Starting|PreFight]HP()` just to make that meaning clear (though that's the meaning I would have naturally inferred).

I feel like this PR is bringing out some major differences in how we approach/interpret things.

Reply:
> > I'm making him acknowledge that he's looking at the current board state
> 
> Hm. I'm not sure that distinction is as clear as you imagine it to be.

Well. I'm looking at a reference to a unit that is (theoretically) on the map somewhere, waiting to act.
My first assumption on talking to that unit is that it has zero knowledge that a combat calculation is taking place. It knows the current board state, and nothing else.
I guess you could make the assumption that the unit is *cloned* for the purposes of combat calculation, but we avoid cloning unless it's necessary.

> if I'm just looking for an accessor (How much health does this guy have, anyway), I'm not going to look closely enough to see that the unit is declared `final` and then make generalizations about what that means about the state of the world.

That's entirely fair. but if you're looking for an accessor, you'll go `params.atta` and Eclipse will happily pop up and point you to `attackerHP`, and all will be right with the world.
Then you say, "Wait, what's the *actual* HP of the unit as it is on the board right now?", so you go to the unit reference.

> Maybe add the utility method with a name like `get[Starting|PreFight]HP()` just to make that meaning clear (though that's the meaning I would have naturally inferred).

I guess I could, but we clearly don't mind having call-chains (see: `unit.model.thing` littered all throughout the codebase)

> I feel like this PR is bringing out some major differences in how we approach/interpret things.

It may also be worth noting that when I move stuff around, I typically have to change a few lines in Muriel. Wally is forced to change much less frequently.
I guess what I'm saying is I already was noticing there are some differences, lol.

Reply:
>  if you're looking for an accessor, you'll go params.atta

Well, maybe, or maybe I'll assume that accessors were named with the normal Java-style `get` prefix and start there. And then I'll be confused because I know units are combatants, so why doesn't this have unit's functions? And then I'll notice that it's a wrapper around a unit and try to type `params.attacker.getUnit`, which isn't there, and then `params.attacker.unit` which still isn't there, and then notice that it's actually been helpfully named `body` for some reason, and throw my keyboard at the ceiling.

Reply:
> Well, maybe, or maybe I'll assume that accessors were named with the normal Java-style `get` prefix and start there.

I don't think that assumption is all that valid. We use bare member variables in most any place that we aren't looking to specifically gain something from using functions.

> And then I'll be confused because I know units are combatants, so why doesn't this have unit's functions? And then I'll notice that it's a wrapper around a unit and try to type `params.attacker.getUnit`, which isn't there, and then `params.attacker.unit` which still isn't there, and then notice that it's actually been helpfully named `body` for some reason, and throw my keyboard at the ceiling.

So what you're saying is you don't like my creative naming schemes?
IDK, something about naming variables the same thing as their types rubs me the wrong way. Seems like you should be naming the variable for the usage *in this instance*, not just aping whatever the type name is.
I don't feel super strong about it, though, since standardization is also good.

Reply:
> I don't think that assumption is all that valid. We use bare member variables in most any place that we aren't looking to specifically gain something from using functions.

Yeah, we have done a terrible job maintaining proper object-oriented encapsulation or any kind of consistent naming scheme. Which is fine for a two-developer side project, but wouldn't fly in the real world. I mean, actually it would because real-world code review practices are often not great, but it certainly doesn't help with handing off code.

> something about naming variables the same thing as their types rubs me the wrong way.

Must be tough being you, because naming variables after object types (typically modified by an adjective, e.g. `myUnit` or `attackingUnit` or `unitReference`) is standard practice almost everywhere. 

> Seems like you should be naming the variable for the usage `in this instance`

Which makes sense, and is what the adjective is for. Naming the `Combatant`'s `Unit` "body" doesn't though, because DP doesn't have a concept of unit bodies, and it just seems like you got bored while typing and decided to be clever. As much as I appreciate creativity, and as much as software design is a creative art, there is still great value in sticking with standard/predictable naming conventions.
I'm not saying you have to change it, because it's clear enough in this small class which is only used in a narrow context, but with something that is used in a larger context it could become confusing.

Reply:
> Yeah, we have done a terrible job maintaining proper object-oriented encapsulation or any kind of consistent naming scheme.

Naming schemes, I'll give you.
Encapsulation exists to prevent access to implementation details. Just adding getters and setters for every variable isn't any better encapsulation than allowing raw member access.
HP, for instance, doesn't have raw access because I wanted to be explicit about exactly what "form" of it you were accessing, and to enforce certain semantics on exactly how it is modified.
CO str/def doesn't have raw access because I wanted to force changes to those stats to use dead reckoning: "I move it up 20, then back down 20" as opposed to "I set it to 120, then back to 100", since the second doesn't play well with multiple modifiers.

I won't say we've done a perfect job of keeping everything hidden that *should* be, but most of the things that are public *are* public because we have reasons to be modifying them.